src/Solvers/preconditioned_conjugate_gradient_solver.jl

```diff
@@ -172,9 +172,9 @@ function solve!(x, solver::PreconditionedConjugateGradientSolver, b, args...)
         iterate!(x, solver, b, args...)
     end
 
-    fill_halo_regions!(x, solver.architecture)
+    fill_halo_regions!(x, solver.architecture) # blocking
 
-    return nothing
+    return x
```

Comment:
if `solve!(x, ...)` returns `x` shouldn't it be named `solve(x, ...)`?

Reply:
The bang `!` indicates that `x` is modified, but does not restrict whether `x` is returned or not. Often the argument is returned so that it displaces at the REPL. We should adopt this practice too I think for things like `compute!`, etc.